service/worker/archiver/activities.go

```diff
@@ -244,11 +258,27 @@ func uploadBlob(ctx context.Context, blobstoreClient blobstore.Client, bucket st
 	return nil
 }
 
+func downloadBlob(ctx context.Context, blobstoreClient blobstore.Client, bucket string, key blob.Key) (*blob.Blob, error) {
+	bCtx, cancel := context.WithTimeout(ctx, blobstoreTimeout)
+	blob, err := blobstoreClient.Download(bCtx, bucket, key)
+	cancel()
+	for err != nil {
+		activity.RecordHeartbeat(ctx)
```

Comment:
Do you think moving heartbeating to separate goroutine makes sense?

Reply:
My first implementation heartedbeated within a goroutine. I had a conversation with Maxim about doing it this way instead of the goroutine approach. The issue with doing it in a separate goroutine is lets say the activity gets stuck - some RPC call is hanging due to a bug. Since the heartbeat is in a separate go routine we will not get a heartbeat timeout. Instead we will have to wait for the context to timeout.